python/cudf/cudf/core/index.py

```diff
@@ -2796,59 +2802,71 @@ def interval_range(
     >>> cudf.interval_range(start=0,end=5)
     IntervalIndex([(0, 0], (1, 1], (2, 2], (3, 3], (4, 4], (5, 5]],
     ...closed='right',dtype='interval')
-
     >>> cudf.interval_range(start=0,end=10, freq=2,closed='left')
     IntervalIndex([[0, 2), [2, 4), [4, 6), [6, 8), [8, 10)],
     ...closed='left',dtype='interval')
-
     >>> cudf.interval_range(start=0,end=10, periods=3,closed='left')
     ...IntervalIndex([[0.0, 3.3333333333333335),
             [3.3333333333333335, 6.666666666666667),
             [6.666666666666667, 10.0)],
             closed='left',
             dtype='interval')
-
     """
     if freq and periods and start and end:
         raise ValueError(
             "Of the four parameters: start, end, periods, and "
             "freq, exactly three must be specified"
         )
+    if isinstance(start or end, dt.datetime) or isinstance(
+        start or end, dt.timedelta
+    ):
+        raise NotImplementedError(
+            "Datetime-like intervals are not yet supported"
+        )
+    if start and isinstance(start, str):
+        raise ValueError(f"start must be numeric, got {start}")
+    if end and isinstance(end, str):
+        raise ValueError(f"end must be numeric, got {start}")
```

Comment:
We'd need to handle things like Pandas timestamps, numpy datetime64, numpy timedelta, etc. here where it's going to get gross quickly. May be a better approach to just assert that `start` / `end` / `freq` are numeric for now and throw a `NotImplementedError` otherwise.